src/_pytest/python.py

```diff
@@ -341,6 +341,50 @@ def reportinfo(self) -> Tuple[Union[py.path.local, str], int, str]:
         return fspath, lineno, modpath
 
 
+# As an optimization, these names are pre-ignored when iterating over
```

Comment:
Hmm, that doesn't seem to contain some of the names listed here, even if I add an empty instance:

```
>>> sorted(set(dir(types.ModuleType)).union(dir(object)).union(dir(object())))
[
    '__class__', 
    '__delattr__', 
    '__dict__', 
    '__dir__', 
    '__doc__', 
    '__eq__', 
    '__format__', 
    '__ge__', 
    '__getattribute__', 
    '__gt__', 
    '__hash__', 
    '__init__', 
    '__init_subclass__', 
    '__le__', 
    '__lt__', 
    '__ne__', 
    '__new__', 
    '__reduce__', 
    '__reduce_ex__', 
    '__repr__', 
    '__setattr__', 
    '__sizeof__', 
    '__str__', 
    '__subclasshook__'
]
```

so `__builtins__`, `__cached__`, `__file__`, `__loader__`, `__module__`, `__name__`, `__package__`, `__spec__` and `__weakref__` would still be missing.

Reply:
Perhaps we should just ignore any `__dunder__` name? Or would that be too much of a breaking change?

Reply:
Thanks for the suggestion @nicoddemus, I applied it now.

> Hmm, that doesn't seem to contain some of the names listed here, even if I add an empty instance:

Right, needed some alterations, but now it's the same. I had to add some names manually, where `types.ModuleType` doesn't have some attributes an empty filesystem module has.

> Perhaps we should just ignore any __dunder__ name? Or would that be too much of a breaking change?

The idea here is a bit more conservative, because I don't think there is any reasonable usecase for having tests with these names, while disallowing arbitrary dunder names might be used so would need a deprecation at least.

Reply:
> The idea here is a bit more conservative, because I don't think there is any reasonable usecase for having tests with these names, while disallowing arbitrary dunder names might be used so would need a deprecation at least.

Yeah I prefer to be conservative here as well.

We might consider adding a deprecation warning for dunder names, but I'm not sure this will buy us much in collection time anyway.

Reply:
(Also just a reminder about adding a CHANGELOG in case you missed it).